fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt

```diff
@@ -60,18 +60,44 @@
   </xsl:template>
 
 
+  <!-- the following template creates an _ms field for thesis advisors -->
```

Comment:
@cdeaneGit, there are some problems with your logic here.  

Like the meeting notes say, nameParts will only be used when someone submits via a form.  It won't do anything about migrated data.  For that reason, you need more logic, and you need to serialize certain things to the field based on that logic.

Specifically, migrated content uses displayForms not nameParts because we can't easily split the string into sections.

For this reason, your code should:

1. Match on mods:mods/mods:name[(mods:role/mods:roleTerm='Thesis advisor') or (mods:role/mods:roleTerm='thesis advisor')]
2. Determine if it has a namePart, and if so serialize this to the field:
`\name\namePart[@type="family", \name\namePart[@type="given", \name\namePart[@type="termsOfAddress"`
3. If not, serialize this to the field:
`\name\displayForm`

@CanOfBees, please let me know if you disagree.

Reply:
http://www.loc.gov/standards/mods/v3/mods-3-5.xsd

****************************************************
*   Top Level Element <name>                 *
*****************************************************         
--><xs:element name="name" type="nameDefinition"/>
<!-- -->
<xs:complexType name="nameDefinition">
<xs:choice>
<!--  this choice is introduced in version 3.5, with the introduction of <etal>.  
      There are two "choices", one without <etal> (the old way) and one with <etal>.
      Choice one. The old way.
-->
<xs:choice minOccurs="0" maxOccurs="unbounded">
  <xs:element ref="namePart"/>
  <xs:element ref="displayForm"/>
  <xs:element ref="affiliation"/>
  <xs:element ref="role"/>
  <xs:element ref="description"/>
</xs:choice>

Because the MODS online schema sows namePart and displayForm at the same level,
I am treating them the same way in the template.

Reply:
@markpbaggett that sounds correct.

Reply:
I agree with the order of operations, putting displayForm last.
Since AdvisorFirstName, AdvisorLastName are required, we always have them with new ingests.
Legacy data:  Use displayForm value.
BUT --  Is there ever a case where the displayForm data is MISSING from legacy data?
Should the otherwise be
<xsl:value-of select="NOT AVAILABLE" />
?

Reply:
Bridger answered my question.  The migration data does not have any case where NOT AVAILABLE would be needed.